### ISSUE TITLE
Hard to reproduce

### ISSUE THREAD

**Issue by dinggd** [PKU-ICST-MIPL/FGCrossNet_ACMMM2019#2]
# 1 Request for your exact testing data
Since the testing results with the provided model on our downloaded data is very different from reported as below:

|     |  mAP |
| --- | --- |
| i2t | 0.545611 |
| i2a | 0.526377 |
| i2v | **0.006164** |
| t2i | 0.553736 | 
| t2a | 0.412829 |
| t2v | **0.006148** |
| a2i | 0.553225 |
| a2t | 0.418043 |
| a2v | **0.006096** |
| v2i | **0.015678** |
| v2t | **0.011916** |
| v2a | **0.012020** |
| i2all | 0.413965 |
| t2all | 0.327447 |
| a2all | 0.326022 |
| v2all | **0.017840** |

Could you please kindly provide the **exact testing data** for a quick test of your model? BaiduDisk, Google Disk or Onedrive at your choice. 

PS: I noticed that the resolution of our downloaded is 360P and yours is 720P, however, some of the videos on youtube can not be downloaded anymore since they either private, deleted or lack the resolution of 720P.

# 2 Dataloader
Should the ```shuffle``` option always be ```False``` for the training dataloader?

https://github.com/PKU-ICST-MIPL/FGCrossNet_ACMMM2019/blob/b92a2aefdc7ae52069cc4353be999ac07c42af20/main.py#L126

If so, all the remaining text, audio, and video frames will never be input to the networking for training since the iterations are based on the batch of images.

And also, during the training of ranking loss, how do you guarantee two positive cross-modal pair and two negative samples?

Thanks!

**Comment by mumuhe** [comment]
> esting results with the provided mo

Thanks for your interest to our work.

For the first question, I think the problem maybe the video data. From the results, I see the results related to video are all bad. So I suggest checking the video data.

For the second question, we construct the training list, so we set the shuffle option False.

**Comment by dinggd** [comment]
> > esting results with the provided mo
> 
> Thanks for your interest to our work.
> 
> For the first question, I think the problem maybe the video data. From the results, I see the results related to video are all bad. So I suggest checking the video data.
> 
> For the second question, we construct the training list, so we set the shuffle option False.

Thanks for the kind reply.

Yes,  we have noticed the performance with regard to the video is extremely low and we assumed the problem might be the provided model, which is **trained with higher(720P)** resolution frames while **testing with lower (360P)** in our case.

We tried to finetune with video data that we can get but nothing improved much. 

Can you please kindly advise how to construct the training list? A sample code is very much appreciated!

Thanks again!

**Comment by mumuhe** [comment]
> > > esting results with the provided mo
> > 
> > 
> > Thanks for your interest to our work.
> > For the first question, I think the problem maybe the video data. From the results, I see the results related to video are all bad. So I suggest checking the video data.
> > For the second question, we construct the training list, so we set the shuffle option False.
> 
> Thanks for the kind reply.
> 
> Yes, we have noticed the performance with regard to the video is extremely low and we assumed the problem might be the provided model, which is **trained with higher(720P)** resolution frames while **testing with lower (360P)** in our case.
> 
> We tried to finetune with video data that we can get but nothing improved much.
> 
> Can you please kindly advise how to construct the training list? A sample code is very much appreciated!
> 
> Thanks again!

We restrict that the 4 instances belong to 3 subcategories, which means two of them belong to the same subcategory.

**Comment by qingjiusanliangsan** [comment]
> > > > esting results with the provided mo
> > > 
> > > 
> > > Thanks for your interest to our work.
> > > For the first question, I think the problem maybe the video data. From the results, I see the results related to video are all bad. So I suggest checking the video data.
> > > For the second question, we construct the training list, so we set the shuffle option False.
> > 
> > 
> > Thanks for the kind reply.
> > Yes, we have noticed the performance with regard to the video is extremely low and we assumed the problem might be the provided model, which is **trained with higher(720P)** resolution frames while **testing with lower (360P)** in our case.
> > We tried to finetune with video data that we can get but nothing improved much.
> > Can you please kindly advise how to construct the training list? A sample code is very much appreciated!
> > Thanks again!
> 
> We restrict that the 4 instances belong to 3 subcategories, which means two of them belong to the same subcategory.

Thanks for your work,I found that there only have the list file of test, could you provide the list file of train and I just need the file of "./list/video/train.txt" ,thanks very much!

**Comment by 284513016** [comment]
> > > esting results with the provided mo
> > 
> > 
> > Thanks for your interest to our work.
> > For the first question, I think the problem maybe the video data. From the results, I see the results related to video are all bad. So I suggest checking the video data.
> > For the second question, we construct the training list, so we set the shuffle option False.
> 
> Thanks for the kind reply.
> 
> Yes, we have noticed the performance with regard to the video is extremely low and we assumed the problem might be the provided model, which is **trained with higher(720P)** resolution frames while **testing with lower (360P)** in our case.
> 
> We tried to finetune with video data that we can get but nothing improved much.
> 
> Can you please kindly advise how to construct the training list? A sample code is very much appreciated!
> 
> Thanks again!

Hello
How can I get the complete test and training video file?
I cannot get the complete YouTube-Birds and YouTube-Cars.
Maybe you have a better video file?
Looking forward to your reply  
Thank you

**Comment by HIT16S** [comment]
@gddingcs I have the same problem as you, so have you solved it? How to generate the train list ?

**Comment by NateLol** [comment]
> @gddingcs I have the same problem as you, so have you solved it? How to generate the train list ?

No actually, I do not know the solution to this and I have moved on.
